crates/editor/src/editor.rs

```diff
@@ -6670,6 +6675,63 @@ impl Editor {
         None
     }
 
+    pub(crate) fn update_semantic_tokens(
+        &mut self,
+        buffer: &Entity<Buffer>,
+        window: &mut Window,
+        cx: &mut Context<Self>,
+    ) {
+        if let Some(sema) = self.semantics_provider.as_ref() {
+            let buffer_id = buffer.read(cx).remote_id();
+            let task = sema
+                .semantic_tokens(buffer.clone(), cx)
+                .map(move |task| task.map(move |tokens| (tokens, buffer_id)));
+
+            self.update_semantic_tokens_task = cx.spawn_in(window, async move |this, cx| {
```

Comment:
Thank you for using a similar approach as other parts (and the tests, by the way!), yet we also need to take the debounce bit in.

https://github.com/zed-industries/zed/blob/1e58fe77d280e76ead3a7b0802342e6d5daa10fa/crates/editor/src/editor.rs#L15529-L15541

for example, or other editor tasks.

Since it's a visual thing, it makes sense to get this debounce into new LSP settings section for semantic highlights.